lib/containers/container_images.pm

```diff
@@ -195,8 +221,14 @@ sub test_zypper_on_container {
         # Remove it
         assert_script_run("$runtime rm refreshed", 120);
         # Verify the image works
-        assert_script_run("$runtime run --rm refreshed-image sh -c 'zypper -v ref | grep \"All repositories have been refreshed\"'", 120);
+        if ($runtime =~ /docker/) {
+            assert_script_run("$runtime run --rm refreshed-image sh -c ' zypper -v ref | grep \"All repositories have been refreshed\"'", 120);
+        }
+        else {
+            assert_script_run("$runtime run --rm refreshed-image zypper -v ref | grep \"All repositories have been refreshed\"", 120);
+        }
```

Comment:
I suggest to extract the common string part and do something like this:

```suggestion
        my $check = 'grep \"All repositories have been refreshed\"';
        my $cmd = "$runtime run --rm refreshed-image";
        assert_script_run($runtime =~ /docker/ ? "$cmd sh -c ' zypper -v ref | $check'" : "$cmd zypper -v ref | $check", 120);
```

Reply:
addressed on ba1bb0c9d along with the commit message. thanks a lot